src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -522,11 +522,10 @@ private void InferTypeArgsFirstPhase(Binder binder, ref HashSet<DiagnosticInfo>
             // SPEC: For each of the method arguments Ei:
             for (int arg = 0, length = this.NumberArgumentsToProcess; arg < length; arg++)
             {
-                var argument = _arguments[arg];
-
-                bool isExactInference = GetRefKind(arg) != RefKind.None;
-
+                BoundExpression argument = _arguments[arg];
                 TypeSymbol target = _formalParameterTypes[arg];
+                bool isExactInference = GetRefKind(arg).IsManagedReference() || argument.Type.IsPointerType() || target.IsPointerType();
```

Comment:
Is there a test for successful inference when the argument type is not a pointer type? Perhaps:
```
static void F<T>(T* a, T b) where T : unmanaged
{
    F(null, b);
}
```

Reply:
Is the check for `argument.Type.IsPointerType()` necessary?

Reply:
No. I'll add one.

---
In reply to: [182085539](https://github.com/dotnet/roslyn/pull/26154#discussion_r182085539) [](ancestors = 182085539)